### PR TITLE
Update CI dependencies and fix tests; generate opset 20

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
1. Update ONNX, ORT version in CI
2. Pin IR version in run_converter_test to 9 because ONNX runtime does not support IRv10 yet. Need a better mechanism to handle the IR version change in tests.
3. Generate opset 20 for onnxscript
4. Skip opset 21 tests
5. Fixes https://github.com/microsoft/onnxscript/issues/1318

### CODE REVIEW

**Comment by thiagocrepaldi** [comment]
Unexpected error:

```bash
2024-04-02T16:33:59.5097958Z [31m[1m_ TestModelSaving.test_input_output_and_initializer_are_not_stored_in_value_info _[0m
2024-04-02T16:33:59.5099409Z [gw1] linux -- Python 3.10.14 /home/runner/work/onnxscript/onnxscript/.nox/test/bin/python
2024-04-02T16:33:59.5101242Z [1m[31m.nox/test/lib/python3.10/site-packages/torch/onnx/_internal/exporter.py[0m:1195: in dynamo_export
2024-04-02T16:33:59.5102596Z     ).export()
2024-04-02T16:33:59.5103515Z [1m[31m.nox/test/lib/python3.10/site-packages/torch/onnx/_internal/exporter.py[0m:950: in export
2024-04-02T16:33:59.5104359Z     graph_module = pre_export_passes(
2024-04-02T16:33:59.5105339Z [1m[31m.nox/test/lib/python3.10/site-packages/torch/onnx/_internal/exporter.py[0m:1261: in pre_export_passes
2024-04-02T16:33:59.5106358Z     module = passes.Modularize(diagnostic_context, module).run()
2024-04-02T16:33:59.5107521Z [1m[31m.nox/test/lib/python3.10/site-packages/torch/onnx/_internal/diagnostics/infra/decorator.py[0m:151: in wrapper
2024-04-02T16:33:59.5108450Z     ctx.log_and_raise_if_error(diag)
2024-04-02T16:33:59.5109567Z [1m[31m.nox/test/lib/python3.10/site-packages/torch/onnx/_internal/diagnostics/infra/context.py[0m:366: in log_and_raise_if_error
2024-04-02T16:33:59.5110608Z     raise diagnostic.source_exception
2024-04-02T16:33:59.5111642Z [1m[31m.nox/test/lib/python3.10/site-packages/torch/onnx/_internal/diagnostics/infra/decorator.py[0m:135: in wrapper
2024-04-02T16:33:59.5112589Z     return_values = fn(*args, **kwargs)
2024-04-02T16:33:59.5113319Z [1m[31m.nox/test/lib/python3.10/site-packages/torch/onnx/_internal/fx/_pass.py[0m:267: in run
2024-04-02T16:33:59.5113915Z     module = self._run(*args, **kwargs)
2024-04-02T16:33:59.5114655Z [1m[31m.nox/test/lib/python3.10/site-packages/torch/onnx/_internal/fx/passes/modularization.py[0m:831: in _run
2024-04-02T16:33:59.5115379Z     root_module_node.add_leaf_node(_LeafNode(fx_node))
2024-04-02T16:33:59.5116218Z [1m[31m.nox/test/lib/python3.10/site-packages/torch/onnx/_internal/fx/passes/modularization.py[0m:701: in __init__
2024-04-02T16:33:59.5116975Z     self._stack_meta = _module_stack_meta_from_node(node)
2024-04-02T16:33:59.5117924Z [1m[31m.nox/test/lib/python3.10/site-packages/torch/onnx/_internal/fx/passes/modularization.py[0m:330: in _module_stack_meta_from_node
2024-04-02T16:33:59.5118806Z     return _ModuleStackMeta(node.meta.get("nn_module_stack"))
2024-04-02T16:33:59.5121524Z [1m[31mE   beartype.roar.BeartypeCallHintParamViolation: Method torch.onnx._internal.fx.passes.modularization._ModuleStackMeta.__init__() parameter nn_module_stack_meta={'L__self___fc1': ("L['self'].fc1", <class 'torch.nn.modules.linear.Linear'>)} violates type hint typing.Union[collections.OrderedDict, typing.Dict[str, typing.Tuple[str, typing.Tuple[str, type]]], NoneType], as dict {'L__self___fc1': ("L['self'].fc1", <class 'torch.nn.modules.linear.Linear'>)}:[0m
2024-04-02T16:33:59.5123841Z [1m[31mE   * Not <class "builtins.NoneType"> or <class "collections.OrderedDict">.[0m
2024-04-02T16:33:59.5124832Z [1m[31mE   * dict key str 'L__self___fc1' value tuple index 1 item type <class 'torch.nn.modules.linear.Linear'> not instance of tuple.[0m
```

**Comment by justinchuby** [comment]
Updated failing tests

File: onnxscript/function_libs/torch_lib/graph_building_test.py [hunk 1]
```diff
@@ -138,6 +140,7 @@ def test_add_initializer_allows_adding_the_same_tensor_twice_using_same_name(sel
         graph.add_initializer("x", x_tensor)
 
 
+@unittest.skipIf(IS_WINDOWS, "dynamo_export not supported in Windows")
```
Comment (thiagocrepaldi) [inline onnxscript/function_libs/torch_lib/graph_building_test.py:143]: dynamo is supposed to be supported on windows on pytorch nightly. if something is failing, we need to track and fix

Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/graph_building_test.py:143]: We are not pulling the latest pytorch. I will add additional pytorch version check for this.

**Comment by justinchuby** [comment]
I will work on this pr and get the rest of the stuff green

File: onnxscript/function_libs/torch_lib/graph_building_test.py (hunk 1 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/graph_building_test.py:143]: Do you know since which version?

Comment (thiagocrepaldi) [reply onnxscript/function_libs/torch_lib/graph_building_test.py:143]: https://github.com/pytorch/pytorch/pull/115969

**Review by justinchuby — APPROVED** [review]

**Pull request merged**